src/main/java/org/sasanlabs/service/vulnerability/xss/reflected/XXEVulnerability.java

```diff
@@ -0,0 +1,288 @@
+        try {
+            InputStream in = request.getInputStream();
+            // disabling external Entities and parameter Entities
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            spf.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+
+            return saveJaxBBasedBookInformation(spf, in, LevelConstants.LEVEL_3);
+        } catch (Exception e) {
+            LOGGER.error(e);
+        }
+        return new ResponseEntity<GenericVulnerabilityResponseBean<Book>>(
+                new GenericVulnerabilityResponseBean<Book>(null, false), HttpStatus.OK);
+    }
+
+    // Protects against XXE. This is the configuration where DOCTYPE declaration is
+    // not required.
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_4,
+            htmlTemplate = "LEVEL_1/XXE",
+            requestMethod = RequestMethod.POST,
+            variant = Variant.SECURE)
+    public ResponseEntity<GenericVulnerabilityResponseBean<Book>> getVulnerablePayloadLevel4(
+            HttpServletRequest request) {
+        try {
+            InputStream in = request.getInputStream();
+            // Disabling DocType. Recommended approach
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setFeature("http://apache.org/xml/features/disallow-doctype-decl", true);
+            spf.setFeature("http://xml.org/sax/features/external-general-entities", false);
+            spf.setFeature("http://xml.org/sax/features/external-parameter-entities", false);
+            spf.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
+
+            return saveJaxBBasedBookInformation(spf, in, LevelConstants.LEVEL_4);
+        } catch (Exception e) {
+            LOGGER.error(e);
+        }
+        return new ResponseEntity<GenericVulnerabilityResponseBean<Book>>(
+                new GenericVulnerabilityResponseBean<Book>(null, false), HttpStatus.OK);
+    }
+
+    // Protects against all XXE attacks. This is the configuration which is needed
+    // secure processing.
+    @AttackVector(
+            vulnerabilityExposed = VulnerabilityType.XXE,
+            description = "XXE_ENABLE_SECURE_PROCESSING")
+    @VulnerableAppRequestMapping(
+            value = LevelConstants.LEVEL_5,
+            htmlTemplate = "LEVEL_1/XXE",
+            requestMethod = RequestMethod.POST,
+            variant = Variant.SECURE)
+    public ResponseEntity<GenericVulnerabilityResponseBean<Book>> getVulnerablePayloadLevel5(
+            HttpServletRequest request) {
+        try {
+            InputStream in = request.getInputStream();
+            // disabling external Entities and parameter Entities
+            SAXParserFactory spf = SAXParserFactory.newInstance();
+            spf.setFeature("http://javax.xml.XMLConstants/feature/secure-processing", true);
```

Comment:
I was checking this flag and seems like by default it is enabled for SAXparser.
https://docs.oracle.com/en/java/javase/13/security/java-api-xml-processing-jaxp-security-guide.html#GUID-88B04BE2-35EF-4F61-B4FA-57A0E9102342. 

Also, don't we need other flags to make this level secure?